### ISSUE TITLE
picard version is null

### ISSUE THREAD

**Issue by reJELIN** [ucagenomix/sicelore-2.1#8]
Dear author,

Thank you very much for sicelor, it's very useful.

I'm working on a HPC and i did install picard. I recently run the step : Select ValideCellBarcode. it did work but even though i specified the build path of picard: 

`picard=<my path>/picard/build/libs/picard.jar`

i still got a **picard version is null**:

`Executing as my user_name@number_of_the_node on Linux 3.10.0-957.27.2.el7.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 17.0.4.1+1-LTS-2; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: Version:null`

The processus seems to through the end but the next step tagbamwithread is not ending after 7 day,(it has now been kill because it overpass the time limit) even though it has already write a 32G bam file. **Is it a regular behavior for this function ?**

Thank you in advance for your answer

**Comment by RainerWaldmann** [comment]
regarding the tagbamwithreads: **how much RAM did you allow ?**  It needs enough RAM to keep the reads of one chromosome in RAM.

**Comment by reJELIN** [comment]
I provided something like 64Go of RAM this time and it seems to go through the whole process but just get stuck at the end :

```
Split fastq by chromosome DONE -> will tag bam 
.......................................................................................................................................................................................................................................................................
.......................................................................................................................................................................................................................................................................
.......................................................................................................................................................................................................................................................................
......................................................................................................................................................................................................................
```

Then it didn't write anything more in the bam file after 24h of running, so i killed the  process myself but that's not very convenient

**Comment by kwglam** [comment]
same issue here. But it's sample dependent. I ran 4 samples and only one of them got stuck..... I have increased the RAM from 300G to 500G, but still got stuck there....

**Comment by reJELIN** [comment]
After the tagbamwithread getting stuck you still get a passedParsedWithSequences.bam. Out of curiosity i ran ComputeConsensus with this bam.

**Comment by kwglam** [comment]
Yes, after tagbamwithread, you would still get a passedParsedWithSequences.bam. But this bam file is truncated. For instance, the last line of this bam in my case is from chr5. I think that's the reason why you didn't see mitochrondrial genes. 
Any idea how to solve this problem??

**Comment by reJELIN** [comment]
Yes, it was my guess also. I didn't find any solution to this problem for the moment. I'm waiting a fix.

**Comment by RainerWaldmann** [comment]
Sorry for the late reply.  Currently, the reads of an entire chromosome are kept in the RAM.  Java programs typically slow down and sometimes almost stop running when RAM is exhausted. However I am a bit surprised that kwglam had trouble with 500G of RAM.  Which genome was it ?  Does it have very huge chromosomes ?
I'll  reduce the RAM requirement of this step.

**Comment by reJELIN** [comment]
Infortunately my try with 300GB was indeed stuck for seven day until the time limit slurm stopped it. Do you think i should try again with 400GB or 500GB ? i'm working on human genome and it is only one sample.

**Comment by yn280** [comment]
Hi @RainerWaldmann , I wonder if by any chance you have made any progress on this? I am stuck facing the same issue - stalling:

"version=2.1
build.date=2023-10-16T12:45:44Z
Writing temporary files to: /nfs4/suffolk/GenomicsB/yn280/240910_10x_usp_5024_ONT/.
parsed Bam -> will split Fastq
Split fastq by chromosome DONE -> will tag bam
....................................................................................................................................................................."

**Comment by mcortes-lopez** [comment]
Hi @RainerWaldmann I was wondering if there is any solution on this or if it's possible to run this step by first splitting the bam and fastq inputs by chromosome? 

I have a file that is bigger than other samples I have preprocess with this pipeline and it gets stuck even when I give it 1000G of memory.

**Comment by reJELIN** [comment]
You could do:

 samtools split --max-split 20000 \
        -d CB \
        --threads 1 \
        -f '/per_cell_bams/%!.bam' \
        --no-PG \
        tagged_bam

samtools index -@ 3 -M /per_cell_bams/*.bam


Then I'm using the tool bam-readcounts by giving a bed file to check some specific positions.

The output is tsv by barcodes, then i'm concatening each tsv into a variant matrix.
